src/js/index.js

```diff
@@ -0,0 +1,59 @@
+import Drink from './Drink.js';
+import { $, $$ } from './utils/dom.js';
+
+class DrinkMachine {
+  constructor() {
+    this.drinks = {
+      americano: new Drink('아메리카노', ['에스프레소', '뜨거운 물']),
+    };
+
+    this.$dispenser = $('.dispenser');
+    this.$clearButton = $('.clear');
+    this.$menuButtons = $$('.menu__button');
+
+    $('.menu').addEventListener('click', this.menuButtonClickEvent);
+    this.$clearButton.addEventListener('click', this.clearDispenser);
+  }
+
+  menuButtonClickEvent = e => {
+    if (this.$dispenser.children.length > 0) return;
+
+    e.target.classList.add('active');
+
+    let template = '<div>컵이 나옵니다.</div>';
+
+    switch (e.target.name) {
+      case 'americano':
+        template += this.drinks.americano.make();
+    }
```

Comment:
이 부분 e.target.name 값만 제대로 들어가 있으면, switch 안 쓰고 e.target.name 값으로 drinks에 접근해도 괜찮을 것 같은데요?! 혹시 switch문을 쓴 이유가 있나요?
```suggestion
    template += this.drinks[e.target.name].make();
```

Reply:
전에 두 접근법이 실행되는 순간이 다르다는 글을 본 적이 있었어요.

`this.drinks.americano` -> 컴파일 시
`this.drinks[americano]` -> 런타임 시

저는 만약에 없는 프로퍼티에 접근해서 오류가 발생했다면 런타임 시보다는 컴파일 시 발생하는게 더 나을 것 같다고 생각해서 주로 `.`을 사용한 접근을 쓰고는 했었는데요, 여기에서도 그거에 익숙해진 나머지 당연히 switch를 사용해야지라고 생각했네요 🫠

지금은 아메리카노 하나라서 괜찮지만, 만약에 음료수의 양이 많아진다면 switch를 사용하지 말아야 할 이유가 생길 것 같아요 👀 소피아가 말씀해주신대로 `[]`를 사용하는 방식으로 바꿔볼게요! 감사합니다 🤗

Reply:
대괄호 접근법을 쓰면 접근하려는 프로퍼티 이름이 런타임 시점에 동적으로 결정되어서 그랬군요! ([객체로 작업하기 - MDN](https://developer.mozilla.org/ko/docs/Web/JavaScript/Guide/Working_with_Objects)) 이 부분까지는 생각해보지 못 했는데, 덕분에 관련해서 한번 훑어보고 생각해볼 수 있었어요. 👍
그런 이유라면 컴파일 과정에서 오류가 발생하는 게 나을 수 있겠네요. 다만 하리 말씀대로 음료 종류가 많아지면 switch가 엄청 길어질 것 같긴 해요. 하리가 더 선호하는 방식으로 해보시면 좋겠어요! 덕분에 저도 공부했어요. 감사합니다!

Reply:
오 두 방식간의 실행되는 순간에 차이가 있군요!
메모 메모...📝